src/ray/raylet/scheduling_queue.cc

```diff
@@ -111,34 +107,48 @@ bool SchedulingQueue::TaskQueue::RemoveTask(const TaskID &task_id,
   return true;
 }
 
-bool SchedulingQueue::TaskQueue::HasTask(const TaskID &task_id) const {
+bool TaskQueue::HasTask(const TaskID &task_id) const {
   return task_map_.find(task_id) != task_map_.end();
 }
 
-const std::list<Task> &SchedulingQueue::TaskQueue::GetTasks() const { return task_list_; }
+const std::list<Task> &TaskQueue::GetTasks() const { return task_list_; }
 
-const ResourceSet &SchedulingQueue::TaskQueue::GetCurrentResourceLoad() const {
+const ResourceSet &TaskQueue::GetCurrentResourceLoad() const {
   return current_resource_load_;
 }
 
+bool ReadyQueue::AppendTask(const TaskID &task_id, const Task &task) {
+  auto &resources = task.GetTaskSpecification().GetRequiredResources();
+  RAY_CHECK(tasks_with_resources_[resources].insert(task_id).second == true);
+  return TaskQueue::AppendTask(task_id, task);
+}
+
+bool ReadyQueue::RemoveTask(const TaskID &task_id, std::vector<Task> *removed_tasks) {
+  if (task_map_.find(task_id) != task_map_.end()) {
```

Comment:
How about calling `TaskQueue::RemoveTask` first, then using the return value for this if check?

Reply:
That's not possible, since `TaskQueue::RemoveTask` will destroy the task/resources and then we can't update the `tasks_with_resources_` map any more.